src/framework/preferences/js/Enactors.js

```diff
@@ -427,4 +427,244 @@ var fluid_2_0_0 = fluid_2_0_0 || {};
+            $(this).css("color", colorToSet);
+            var backgroundColor = initialColors[index].backgroundColor;
+            backgroundColor[2] = Math.round(backgroundColor[2] * (1 - times)) + "";
+            if (backgroundColor[3] === undefined) {
+                var backgroundColorToSet = "rgb(" + backgroundColor[0] + ", " + backgroundColor[1] + ", " + backgroundColor[2] + ")";
+            }
+            else {
+                var backgroundColorToSet = "rgba(" + backgroundColor[0] + ", " + backgroundColor[1] + ", " + backgroundColor[2] + ", " + backgroundColor[3] + ")";
+            }
+            $(this).css("background-color", backgroundColorToSet);
+        });
+    };
+
+    fluid.prefs.enactor.blueColorFilter.addMutationObserver = function (applyFilter) {
+        // Uses document ready, because the script is added before the body
+        $(document).ready(function (){
+            var isSafari = /^((?!chrome|android).)*safari/i.test(navigator.userAgent);
+            if (isSafari) {
+                var observer = new WebKitMutationObserver(function(mutations) {
+                    mutations.forEach(function (mutation) {
+                        if (mutation.type === "childList") {
+                            // Changes in HTMLHtmlElement can be just hovering over elements, which should not trigger applyFilter
+                            if (changedDOMFlag === false && mutation.target != "[object HTMLHtmlElement]") {
+                                changedDOMFlag = true;
+                                applyFilter();
+                            }
+                        }
+                    });
+                    changedDOMFlag = false;
+                });
+            }
+            else {
+                var observer = new MutationObserver(function(mutations) {
+                    mutations.forEach(function (mutation) {
+                        if (mutation.type === "childList") {
+                            // Changes in HTMLHtmlElement can be just hovering over elements, which should not trigger applyFilter
+                            if (changedDOMFlag === false && mutation.target != "[object HTMLHtmlElement]") {
+                                changedDOMFlag = true;
+                                applyFilter();
+                            }
+                        }
+                    });
+                    changedDOMFlag = false;
+                });
+            }
+
+            var target = document.body;
+            var config = {
+                childList : true,
+                subtree : true
+            };
+
+            observer.observe(target, config);
+        });
+    };
+
+    /* Add a mutationObserver to track when the UI panel in its responsive version is opened.
+       It then adds and removes a class to the hide/show button to style it properly. */
+    $(document).ready(function (){
```

Comment:
Refactor this chunk of code into one or more infusion components. You may wanna consider to use context awareness (http://docs.fluidproject.org/infusion/development/ContextAwareness.html) to check if Safari is used and attach proper grade(s) in different scenarios.